### ISSUE TITLE
Topics: too many emits

### ISSUE THREAD

**Issue by NicolasSiver** [NodeBB/NodeBB#3793]
Reproducing:

- Create topic
- Have several embeds in the same time
- Page enters some sort of loop

Console Output:

```
27/10 19:18 [55136] - warn: Flooding detected! Calls : 101, Duration : 5401
27/10 19:18 [55136] - warn: [socket.io] Too many emits! Disconnecting uid : 1. Events : topics.loadMore,topics.markAsRead,topics.loadMore,topics.markAsRead,topics.loadMore,topics.markAsRead,topics.loadMore,topics.markAsRead,topics.loadMore,topics.markAsRead,topics.loadMore,topics.markAsRead,topics.loadMore,topics.markAsRead,topics.loadMore,topics.markAsRead,topics.loadMore,topics.markAsRead,topics.loadMore,topics.markAsRead
```

By visiting pages with just youtube embeds, everything is OK.

**Comment by NicolasSiver** [comment]
FYI, Plugin is ready to be released: https://github.com/NicolasSiver/nodebb-plugin-ns-embed, but I can't because of the issue.

**Comment by barisusakli** [comment]
Sounds like infinite scroll is triggering and calling `topics.loadMore` and then `topics.markAsRead`

https://github.com/NodeBB/NodeBB/blob/master/public/src/client/topic/posts.js#L204-L206

Infinite scroll triggers on a window scroll event. Are you doing anything that would cause window scroll?

**Comment by NicolasSiver** [comment]
No, just on parse phase insert several iframes.
I have created topic with all 5 embeds that plugin supports, and NodeBB is going crazy.
Don't understand how several embeds can do such things ;)

**Comment by NicolasSiver** [comment]
Is there any solution? 
I really would like to release plugin.

**Comment by barisusakli** [comment]
When you remove the embeds does it work properly? Maybe those embeds are somehow causing window scroll events not sure without seeing a live example.

**Comment by NicolasSiver** [comment]
Hm, I could try all embeds one by one.
Maybe `vine` or `coub` can do such things (don't know for what...)

Twitch I already disabled, because they do not support `autoPlay = false`...

Will provide feedback today's evening.

**Comment by NicolasSiver** [comment]
Ok, something is wrong...

https://github.com/NodeBB/NodeBB/blob/v0.8.x/public/src/client/topic/posts.js#L250

I have added console log (`load more`), and by scrolling page, It creates so many calls, especially when you scroll page top...

Here is an example, when you scroll general page:

<img width="86" alt="screen shot 2015-10-30 at 10 18 46 pm" src="https://cloud.githubusercontent.com/assets/135263/10861411/5056490a-7f54-11e5-997b-bd7560fa1357.png">

And here is an example of scroll of the page with single embed: Youtube 

<img width="110" alt="screen shot 2015-10-30 at 10 20 14 pm" src="https://cloud.githubusercontent.com/assets/135263/10861416/78b33534-7f54-11e5-98c6-849266479e7b.png">

I have received one `load more` when just entered topic, and 50 updates, while scrolled it back to top...

**Comment by NicolasSiver** [comment]
For embeds I'm using responsive iFrames

https://github.com/NicolasSiver/nodebb-plugin-ns-embed/blob/master/style/forum.less

**Comment by NicolasSiver** [comment]
Maybe a problem with infinitescroll script? - https://github.com/NodeBB/NodeBB/blob/master/public/src/client/infinitescroll.js
Would you consider to use 3-rd party library for such task? - https://github.com/cubiq/iscroll/

**Comment by NicolasSiver** [comment]
Reproduce:

1. `git clone https://github.com/NicolasSiver/nodebb-plugin-ns-embed.git`
2. npm link
3. cd /to/NodeBB/
4. npm link nodebb-plugin-ns-embed
5. Activate plugin, and install default rules (one button click)
6. Create topic with such content:

```
The path of the righteous man is beset on all sides by the iniquities of the selfish and the tyranny of evil men. 

https://youtu.be/mCp5As-Oo6E

https://vimeo.com/channels/staffpicks/139034944

https://vine.co/v/e605UuTVLg6

http://coub.com/view/5u5n1

http://www.twitch.tv/teawrex
```

7. Now play a bit with a topic: scroll, etc. (my CPU boils...)

<img width="642" alt="screen shot 2015-11-05 at 8 34 00 pm" src="https://cloud.githubusercontent.com/assets/135263/10986952/9fe0bbe8-83fc-11e5-8a8e-e779d1c6e7f0.png">

**Comment by barisusakli** [comment]
Try changing this https://github.com/NodeBB/NodeBB/blob/v0.8.x/public/src/client/topic/posts.js#L260 to `setTimeout(done, 200);` see if it makes a difference.

**Comment by NicolasSiver** [comment]
@barisusakli 200 ms delay will not help much, such action just postpone callback on 200 ms.
`loadMore` is just triggered more than once.
I have recorded video (Download for better quality) - https://www.dropbox.com/s/4d1yuqa4zfwto5k/emits.mov?dl=0

When you scroll top, it triggers `loadMore` - https://github.com/NodeBB/NodeBB/blob/v0.8.x/public/src/client/topic/posts.js#L249

**Comment by barisusakli** [comment]
It should be better than calling it as fast as it can, did you try it?

**Comment by barisusakli** [comment]
Delaying the calling of the callback results in less calls to loadMore because of this https://github.com/NodeBB/NodeBB/blob/v0.8.x/public/src/client/infinitescroll.js#L44

**Comment by NicolasSiver** [comment]
But, delaying something isn't a fix... Proper logic will be triggered just once...
Logic of infinite scroll just doesn't work well.

**Comment by barisusakli** [comment]
Is the plugin not compatible with 0.9.0? I don't see it in the plugins dropdown in the ACP after installing and activating.

**Comment by NicolasSiver** [comment]
Correct, - `0.8.x`

**Comment by barisusakli** [comment]
Are you still having this issue after #3862 is fixed?

**Comment by NicolasSiver** [comment]
#3862 isn't resolved this issue.
